src/routes/blogs.rs

```diff
@@ -111,42 +108,43 @@ pub fn create(
         );
     }
 
-    if errors.is_empty() {
-        let blog = Blog::insert(
-            &*conn,
-            NewBlog::new_local(
-                slug.clone(),
-                form.title.to_string(),
-                String::from(""),
-                Instance::get_local()
-                    .expect("blog::create: instance error")
-                    .id,
-            )
-            .expect("blog::create: new local error"),
-        )
-        .expect("blog::create:  error");
+    if !errors.is_empty() {
+        return render!(blogs::new(&rockets.to_context(), &*form, errors)).into();
+    }
 
-        BlogAuthor::insert(
-            &*conn,
-            NewBlogAuthor {
-                blog_id: blog.id,
-                author_id: user.id,
-                is_owner: true,
-            },
+    let blog = Blog::insert(
+        &*conn,
+        NewBlog::new_local(
+            slug.clone(),
+            form.title.to_string(),
+            String::from(""),
+            Instance::get_local()
+                .expect("blog::create: instance error")
+                .id,
         )
-        .expect("blog::create: author error");
+        .expect("blog::create: new local error"),
+    )
+    .expect("blog::create:  error");
 
-        Ok(Flash::success(
-            Redirect::to(uri!(details: name = slug.clone(), page = _)),
-            &i18n!(intl, "Your blog was successfully created!"),
-        ))
-    } else {
-        Err(render!(blogs::new(&rockets.to_context(), &*form, errors)))
-    }
+    BlogAuthor::insert(
+        &*conn,
+        NewBlogAuthor {
+            blog_id: blog.id,
+            author_id: user.id,
+            is_owner: true,
+        },
+    )
+    .expect("blog::create: author error");
+
+    Flash::success(
+        Redirect::to(uri!(details: name = slug.clone(), page = _)),
+        &i18n!(intl, "Your blog was successfully created!"),
+    )
+    .into()
```

Comment:
So if I read this correctly, the Rust compiler infers the type from the return type of the function, allowing `into()` to function as expected?

Even though I should probably expect that from Rust, that's still pretty neat.